com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -382,7 +383,8 @@ public static void NetworkHide(List<NetworkObject> networkObjects, ulong clientI
 
         private void OnDestroy()
         {
-            if (NetworkManager != null && NetworkManager.IsListening && NetworkManager.IsServer == false && IsSpawned)
+            if (NetworkManager != null && NetworkManager.IsListening && NetworkManager.IsServer == false && IsSpawned
+                && (IsSceneObject == null || (IsSceneObject != null && IsSceneObject.Value != true) ))
```

Comment:
I know it was like this, but do we need IsSceneObject to be tri-state (null / true / false)?  Can we not say that a thing is or is not a scene object?

Reply:
I would love to make that change, but I have been desperately trying to keep this PR specific to the epic.  The only additional changes made in this PR are specific to potential bugs or new bugs introduced by the changes (which there have been several unit tests and related support classes that I have had to make some adjustments to).

What I can do for you is generate a debug log that will show what the value of this property is at varying stages in scene synchronization and level loading to see if there are any remaining conditions where we need an "unassigned value" to mean "it is a scene placed object".  If not, then after this PR and its unit test PR-1030 and PR-967 are merged I could put together another PR that makes that change?

Reply:
+1, I like keeping changes specific to a feature work / epic.